nginx-proxy.conf

```
@@ -12,6 +12,13 @@ map $http_x_forwarded_host $proxy_x_forwarded_host {
 
 proxy_set_header X-Forwarded-Host $proxy_x_forwarded_host;
 
+map $http_govuk_request_id $govuk_request_id {
+  default $http_govuk_request_id;
+  ''      "$pid-$msec-$remote_addr-$request_length";
```

```suggestion
  ''      $request_id;
```

Since govuk_request_id is not something we expect to be parsed, we may as well just use nginx's built in type to keep things simple. This can help signify the format is not special and that this is not code we need to keep in sync with the production nginx perl code.

---

We have now enabled the env var in the `helm-charts` [#1190](https://github.com/alphagov/govuk-helm-charts/pull/1190) so the logs are available in dev environment, in json format.

---

Ah so you'd not expect people to use this without using the production JSON logs? This is information that shouldn't be in dev logs?

It'd be good to include this context in the commit message. Do we want to add a "how-to" for using it: https://github.com/alphagov/govuk-docker/blob/main/docs/how-tos.md ?